js/main.js

```diff
@@ -12,17 +14,33 @@ var $unorderedList = document.querySelector('#entries-list');
 
 function submitHandler(event) {
   event.preventDefault();
-  var dataPoint = {};
-  for (let i = 0; i < $formInputs.length; i++) {
-    const inputName = $formInputs[i].getAttribute('name');
-    dataPoint[inputName] = $formInputs[i].value;
+  if (data.editing === null) {
+    var dataPoint = {};
+    for (let i = 0; i < $formInputs.length; i++) {
+      const inputName = $formInputs[i].getAttribute('name');
+      dataPoint[inputName] = $formInputs[i].value;
+    }
+    dataPoint.entryId = data.nextEntryId;
+    data.nextEntryId += 1;
+    data.entries.unshift(dataPoint);
+    $unorderedList.prepend(renderEntry(dataPoint));
+  } else if (data.editing !== null) {
+    var editEntry = data.entries[data.entries.length - data.editing.entryId];
```

Comment:
make sure you find the entry by specific id, this math won't work once you start having the ability to delete entries since the ids will no longer correspond in the same way to the indexes of the array.